File: 0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.java
class Solution {
    public int strStr(String haystack, String needle) {
        int left = 0;
        int needleLength = needle.length();
        while(left + needleLength - 1 < haystack.length()){
            String sub = haystack.substring(left, left + needleLength);
            if(sub.equals(needle)){
                return left;
            } else {
                left ++;
                
            }
        }
        return -1;
    }
}

File: 0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.py
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        left = 0
        needleLength = len(needle)
        while left + needleLength - 1 < len(haystack):
            sub = haystack[left : left + needleLength]
            if sub == needle:
                return left
            else:
                left += 1
        return -1